### PR TITLE
Remove shortcut from GroupBy.apply

### LINKED ISSUE #834: Understanding Groupbyapply
Hello there, firstly thank you for such an amazing package that bridges the gap between Pandas and PySpark. 
I started using koalas approximately 1 week back and everything was intuitive till the time i stumbled upon `koalas.Groupby.Apply`. 

Code:
```
if __name__ == '__main__':

        ks_df = ks.DataFrame(features_data)
        ks_df_info_abt_train = ks_df.groupby(['div_nbr', 'store_nbr']).apply(_koalas_train)
        
        def _koalas_train(frame):
                  out_frame = frame.copy()
                  out_frame = frame['trans_type_value'].sum()
                  return out_frame
```
Here features_data is a  pd.Dataframe.

Output from Koalas.Groupby.Apply:

Screen Shot 2019-09-27 at 2 27 26 PM

Output from Pandas.Groupby.Apply:
Screen Shot 2019-09-27 at 2 30 13 PM

As you can see, the output from pandas Groupby apply is as expected, but the output from Koalas Groupby apply is not right. Could you guid me towards the right direction by pointing out any logical mistake that i might have made or anything else. 
Thank you once again.

Koalas version - 0.18.0 
Pandas version - 0.23.4
PySpark -  2.4.3

### PR DESCRIPTION
**Pull request by harupy**
The current `GroupBy.apply` uses a shortcut (proposed in #646) when schema inference is triggered and the number of records is less than or equal to `compute.shortcut_limit` (1000 by default), but this might cause issues such as #834.

This PR proposes to remove the shortcut and make `GroupBy.apply` only infer the schema.

Closes #834

### CODE REVIEW

**Comment by harupy** [comment]
Is it okay to remove shortcut tests in `test_apply` and `test_apply_with_new_dataframe` ([code](https://github.com/databricks/koalas/blob/1462291490c4a3e084ece28edaef92721bce14fd/databricks/koalas/tests/test_groupby.py#L544-#622))?

**Comment by ueshin** [comment]
@harupy Thanks for working on this!
I think we can remove the shortcut tests, but I'd like to ask you to add a test for the case the result was explicitly wrong instead. I guess the example in your comment (https://github.com/databricks/koalas/issues/834#issuecomment-536829367) should be good.

**Comment by harupy** [comment]
@ueshin Thanks! I'll add it.

**Comment by harupy** [comment]
```
______________ [doctest] databricks.koalas.groupby.GroupBy.apply _______________
793            c0  c1  c2
794         0  bb   6  10
795         1  aa   3  10
796         2  aa   4  12
797 
798         You can omit the type hint and let Koalas infer its type.
799 
800         >>> def plus_min(x):
801         ...    return x + x.min()
802         >>> g.apply(plus_min)  # doctest: +NORMALIZE_WHITESPACE
Differences (unified diff with -expected +actual):
    @@ -1,4 +1,4 @@
         A  B   C
    +2  bb  6  10
     0  aa  2   8
     1  aa  3  10
    -2  bb  6  10
```
The doctest failed. We might need `sort_index()`.

**Comment by ueshin** [comment]
Yup, that's fine. Please add `sort_index()`.

File: databricks/koalas/groupby.py [hunk 1]
```diff
@@ -838,17 +838,16 @@ def apply(self, func):
...
             kdf = DataFrame(pdf)
             return_schema = kdf._sdf.schema
-            if len(pdf) <= limit:
-                return kdf
 
         sdf = self._spark_group_map_apply(
-            func, return_schema, retain_index=should_infer_schema)
+            lambda pdf: pdf.groupby(input_groupnames).apply(func),
+            return_schema,
+            retain_index=should_infer_schema)
```
Comment (HyukjinKwon) [inline databricks/koalas/groupby.py]: This fix LGTM

**Comment by HyukjinKwon** [comment]
Merged. Thanks!

**Comment by harupy** [comment]
@HyukjinKwon Thanks for supporting!

**Pull request merged**